### ISSUE TITLE
No negative attributions when using IG with BERT embeddings and 1-layer RNN

### ISSUE THREAD

**Issue by itsmemala** [pytorch/captum#1075]
I have a single layer RNN model that takes BERT token embeddings as input. When calculating attributions using Integrated Gradients, I don't seem to get any negative attributions at all. But with Occlusion I do get both positive and negative attributions for the same inputs. Any idea what could be causing this?

**Comment by craymichael** [comment]
Apologies for the delay.

This can happen depending on the model, target, baseline, and whether you are looking at signed or absolute attributions.

First, make sure you are not taking absolute values during post-processing. Some examples normalize token importances using absolute values for visualization, which removes sign information.

Second, IG signs are relative to the chosen baseline. If the path from the baseline embedding to the input embedding mostly increases the selected class score, the summed token attributions can all be positive. Occlusion can still produce negative values because it measures a different perturbation: replacing a token and observing the output change.

I would check:

```python
model.eval()
```

and compare:

```python
F(input) - F(baseline)
attributions.sum()
```

If that output difference is positive and the model uses a final hidden state, all-positive token attributions may be plausible. Trying a PAD-token baseline, mean embedding baseline, or attributing logits instead of probabilities can help determine whether the behavior is baseline-specific.

This should answer the question, so I'll close this for now. Please feel free to reply with any follow-up questions.
